adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java

```diff
@@ -482,4 +487,45 @@ public FareRule getFareRule(final String fareId, final String routeId, final Str
         return fareRuleCollection.get(FareRule.getFareRuleMappingKey(fareId, routeId, originId, destinationId,
                 containsId));
     }
+
+    /**
+     * Add a shape point to a shape. A shape is a list of shape points whereas a ShapePoint represents a row from
+     * shapes.txt to this. Return the entity added to the repository if not null, otherwise returns null.
+     *
+     * @param newShapePoint the internal representation of a row from shapes.txt to be added to the repository.
+     * @return the entity added to the repository if not null, otherwise returns null.
+     * @throws IllegalArgumentException if the shape point passed as argument is null
+     */
+    @Override
+    public ShapePoint addShapePoint(final ShapePoint newShapePoint) throws IllegalArgumentException {
+        if (newShapePoint != null) {
+            final String shapeId = newShapePoint.getShapeId();
+            if (shapePerIdShapePtSequence.containsKey(shapeId)) {
+                if (shapePerIdShapePtSequence.get(shapeId).containsKey(newShapePoint.getShapePtSequence())) {
+                    return null;
+                }
+                shapePerIdShapePtSequence.get(shapeId).put(newShapePoint.getShapePtSequence(), newShapePoint);
+            } else {
+                final TreeMap<Integer, ShapePoint> innerMap = new TreeMap<>();
```

Comment:
Nit:

```suggestion
                final Map<Integer, ShapePoint> innerMap = new TreeMap<>();
```

Reply:
This requires to cast innerMap as a TreeMap at line 513511 of this same file. Is it advisable?
See:
https://github.com/MobilityData/gtfs-validator/blob/7dcadaa7cc7d71cc04ecb6518d9ea72e26e10792/adapter/repository/in-memory-simple/src/main/java/org/mobilitydata/gtfsvalidator/db/InMemoryGtfsDataRepository.java#L511